core/src/desul/atomics/Compare_Exchange_OpenMP.hpp

```diff
@@ -8,7 +8,18 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #ifndef DESUL_ATOMICS_COMPARE_EXCHANGE_OPENMP_HPP_
 #define DESUL_ATOMICS_COMPARE_EXCHANGE_OPENMP_HPP_
 #include "desul/atomics/Common.hpp"
-#include<cstdio>
+#include <cstdio>
+#include <omp.h>
+
+static constexpr bool omp_on_host() { return true; }
```

Comment:
Possibly prefix with `desul_`, i.e.
```suggestion
static constexpr bool desul_omp_on_host() { return true; }
```

Reply:
this all needs to go into the desul::Impl namespace.